test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -382,3 +382,54 @@ timestamp_manager::validate_prepare_timestamp(session_simulator *session, uint64
 
     return (0);
 }
+
+/*
+ * Validate the durable timestamp. The constraints on the durable timestamp are:
+ * - Durable timestamp should not be specified for non-prepared transaction.
+ * - Commit timestamp is required before setting a durable timestamp.
+ * - The durable timestamp should not be less than the oldest timestamp.
+ * - The durable timestamp should not be less than or equal to the stable timestamp.
+ * - The durable timestamp should not be less than the commit timestamp.
+ */
+int
+timestamp_manager::validate_session_durable_timestamp(
+  session_simulator *session, uint64_t durable_ts)
+{
+    /* Durable timestamp should not be specified for non-prepared transaction. */
+    if (!session->is_txn_prepared())
+        WT_SIM_RET_MSG(
+          EINVAL, "durable timestamp should not be specified for non-prepared transaction");
+
+    /* Commit timestamp is required before setting a durable timestamp. */
+    if (!session->has_first_commit_timestamp())
+        WT_SIM_RET_MSG(EINVAL, "commit timestamp is required before setting a durable timestamp");
+
+    /* The durable timestamp should not be less than the oldest timestamp. */
+    connection_simulator *conn = &connection_simulator::get_connection();
+    if (conn->has_oldest_ts()) {
+        uint64_t oldest_ts = conn->get_oldest_ts();
+        if (durable_ts < oldest_ts)
+            WT_SIM_RET_MSG(EINVAL,
+              "commit timestamp (" + std::to_string(durable_ts) +
+                ") is less than the oldest timestamp (" + std::to_string(oldest_ts) + ")");
+    }
+
+    /* The durable timestamp should not be less than or equal to the stable timestamp. */
+    if (conn->has_stable_ts()) {
+        uint64_t stable_ts = conn->get_stable_ts();
+        if (durable_ts <= stable_ts)
+            WT_SIM_RET_MSG(EINVAL,
+              "commit timestamp (" + std::to_string(durable_ts) +
+                ") must be after the stable timestamp (" + std::to_string(stable_ts) + ")");
```

Comment:
Np: In the other error messages, you used "is less than..." and here "must be...". We could try to use the same syntax for all error messages for consistency.

Reply:
Updated in [c7ba0e3](https://github.com/wiredtiger/wiredtiger/pull/8398/commits/c7ba0e308f36aa6ff7309125383e4abb058bdf13). 
Good catch!